src/table/table-role/grid-navigation.tsx

```diff
@@ -92,17 +100,30 @@ class GridNavigationProcessor {
   }
 
   public refresh() {
-    if (this._table) {
-      // Update focused cell indices in case table rows, columns, or firstIndex change.
-      if (this.focusedCell) {
-        this.focusedCell = this.findFocusedCell(this.focusedCell.element);
+    // Timeout ensures the newly rendered content elements are registered.
+    setTimeout(() => {
```

Comment:
Would [`flushSync`](https://react.dev/reference/react-dom/flushSync) be useful here to avoid timeouts?
> Call `flushSync` to force React to **flush any pending work** and update the DOM synchronously.

Reply:
It would be neat to have flushAsync of some sort to run the update when ready but not forcing anything. The refresh is called with every render and the flushSync can cause significant performance impact.